src/lib/y2network/connection/connection.rb

```diff
@@ -0,0 +1,20 @@
+require "y2network/interface"
+
+module Y2Network
+  module Connection
+    class Connection
+      # @return [Interface]
+      attr_accessor :interface
+      # @return [String] Bootproto (static, dhcp, none)
+      attr_accessor :bootproto
+      # @return [IPAddr]
+      attr_accessor :ip_address
```

Comment:
i don't think this is basic attribute. The most often you need to distinguish whether the interface was configured dynamically or statically and nothing more. But the connection object can provide a kind of configuration descriptor which can contain some details (like ip address, ...)